src/writer/writer.rs

```diff
@@ -104,22 +104,18 @@ impl<T: Write> Writer<T> {
 				write!(self.stream, "BITMAP\n");
 
 				for y in 0 .. map.height() {
-					let mut value: u64 = 0;
-
-					for x in 0 .. map.width() {
-						value <<= 1;
-						value  |= if map.get(x, y) { 1 } else { 0 };
-					}
-
-					value <<= 8 - (map.width() % 8);
-
-					let hex = format!("{:X}\n", value);
-
-					if (hex.len() - 1) % 2 != 0 {
-						write!(self.stream, "0");
+					for x_char in 0 .. (map.width() + 3) / 4 {
```

Comment:
Where does the `3` come from? Are you sure this is not fixing it for the font you're using and breaking it for every other font?

Feels like there's definitely an issue tho.

Reply:
`(x + 3) / 4` is how I do the integer equivalent to `ceil(x / 4)`; one needs a number of hex characters (4 bits each) that is the number of bits divided by four, rounded up.

Reply:
Could you add a comment clarifying that please? Or do the integer/float/integer dance, I'd rather have more clarity than more performance in this case :panda_face: